File: frontend/src/features/account/api/system-settings.ts
import { apiClient } from '@/lib/api-client';
import type {
  MerchantSettlementMethod,
  MerchantSettlementNetwork,
} from '@/features/account/api/merchant-profile';

export const merchantSettlementMethods = ['bank', 'alipay', 'usdt'] as const;
export const merchantSettlementNetworks = ['TRC20', 'ERC20', 'BEP20', 'POLYGON'] as const;

export interface MerchantSettlementSettings {
  enabledMethods: MerchantSettlementMethod[];
  enabledNetworks: MerchantSettlementNetwork[];
  updatedAt: string;
}

export interface SystemFinanceSettings {
  withdrawalMinimumUsd: string;
  withdrawalFeePercent: string;
  platformFeePercent: string;
}

export interface SystemSettings {
  registrationEnabled: boolean;
  finance: SystemFinanceSettings;
  settlement: MerchantSettlementSettings;
  updatedAt: string;
}

export interface SystemSettingsDraft {
  registrationEnabled: boolean;
  finance: SystemFinanceSettings;
  settlement: Pick<MerchantSettlementSettings, 'enabledMethods' | 'enabledNetworks'>;
}

export function getAdminSystemSettings(signal?: AbortSignal): Promise<SystemSettings> {
  return apiClient.get<SystemSettings>('/admin/system-settings', { signal });
}

export function updateAdminSystemSettings(draft: SystemSettingsDraft): Promise<SystemSettings> {
  return apiClient.put<SystemSettings, SystemSettingsDraft>('/admin/system-settings', draft);
}

export function getMerchantSettlementSettings(
  signal?: AbortSignal,
): Promise<MerchantSettlementSettings> {
  return apiClient.get<MerchantSettlementSettings>('/merchant/settlement-settings', { signal });
}

File: backend/src/domain/system_settings.rs
use jiff::Timestamp;

use super::{MerchantSettlementMethod, MerchantSettlementNetwork};

#[derive(Clone, Debug, Eq, PartialEq)]
pub struct MerchantSettlementSettings {
    pub enabled_methods: Vec<MerchantSettlementMethod>,
    pub enabled_networks: Vec<MerchantSettlementNetwork>,
    pub updated_at: Timestamp,
}

#[derive(Clone, Debug, Eq, PartialEq)]
pub struct SystemFinanceSettings {
    pub withdrawal_minimum_microusd: i64,
    pub withdrawal_fee_bps: i32,
    pub platform_fee_bps: i32,
}

#[derive(Clone, Debug, Eq, PartialEq)]
pub struct SystemSettings {
    pub registration_enabled: bool,
    pub finance: SystemFinanceSettings,
    pub settlement: MerchantSettlementSettings,
    pub updated_at: Timestamp,
}
